indico/modules/categories/controllers/display.py

```diff
@@ -167,3 +174,104 @@ def _process(self):
                            Category.chain_titles)
                  .limit(10))
         return jsonify_data(categories=[_serialize_category(c, with_path=True) for c in query], flash=False)
+
+
+class RHDisplayCategory(RHDisplayCategoryBase):
+    def _process(self):
+        past_threshold = self.now - relativedelta(months=1, day=1, hour=0, minute=0, second=0, microsecond=0)
+        future_threshold = self.now + relativedelta(months=1, day=1, hour=0, minute=0, second=0, microsecond=0)
+        event_query = Event.query.with_parent(self.category).order_by(Event.start_dt.desc())
+        past_event_query = event_query.filter(Event.start_dt < past_threshold)
+        future_event_query = event_query.filter(Event.start_dt >= future_threshold)
+        current_event_query = event_query.filter(Event.start_dt >= past_threshold,
+                                                 Event.start_dt < future_threshold)
+        events = current_event_query.filter(Event.start_dt < future_threshold).all()
+        preload_events(events, persons=True)
+        events_by_month = self.group_by_month(events)
+
+        future_event_count = future_event_query.count()
+        past_event_count = past_event_query.count()
+
+        show_past_events = (self.category.id in session.get('fetchPastEventsFrom', set()) or
+                            (session.user and session.user.settings.get('show_past_events')))
+        if show_past_events:
+            past_events = past_event_query.all()
```

Comment:
Actually.. let's load them with AJAX all the time. Certain categories are huge (8k events in 3285 for example) and rendering them all would be super slow. Probably even rendering them from the AJAX request will be slow :/

might need to consider ignoring the user preference if there are more than a certain number of past events, but let's see how slow displaying them is